### PR TITLE
Add an error message when gpu_targets is not set

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Review by TedThemistokleous — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -133,6 +133,7 @@ jobs:
           -DROCM_ENABLE_GH_ANNOTATIONS=On \
           -DCLANG_TIDY_DEPEND_ON_TARGET=Off \
           -DCLANG_TIDY_CACHE=/data/tidy-cache \
+          -DGPU_TARGETS=gfx908 \
```
Comment (umangyadav) [inline .github/workflows/ci.yaml:136]: Does this mean ROCM-Ubuntu server has gfx908 ? 
I thought it was CPU only server.

Comment (TedThemistokleous) [reply .github/workflows/ci.yaml:136]: I think its needed for this issue with targets.

Comment (pfultz2) [reply .github/workflows/ci.yaml:136]: No, it doesnt have any gpu, but we need to build for at least one target to get build/tidy coverage. We could add multiple targets to this as well.

**Review by umangyadav — APPROVED** [review]

File: src/targets/gpu/CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,10 @@
 # ####################################################################################
 
 list(APPEND CMAKE_PREFIX_PATH /opt/rocm)
+find_package(hip)
+if(NOT GPU_TARGETS)
+    message(FATAL_ERROR "HIP package is broken and has no GPU_TARGETS, please pass -DGPU_TARGETS=$(op/rocm/bin/rocminfo | grep -o -m1 'gfx.*') to cmake to build for your gpu.")
```
Comment (causten) [inline src/targets/gpu/CMakeLists.txt:28]: `/opt` not `op`

**Review by causten — CHANGES_REQUESTED** [review]

**Review by umangyadav — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

**Review by causten — APPROVED** [review]

**Pull request merged**